### ISSUE TITLE
Testing fails on 32-bit builds of cartopy

### ISSUE THREAD

**Issue by ghisvail** [SciTools/cartopy#799]
The test case `TestLambertAzimuthalEqualArea` fails on a 32-bit build of `cartopy` with the following error:

```
======================================================================
FAIL: test_default (cartopy.tests.crs.test_lambert_azimuthal_equal_area.TestLambertAzimuthalEqualArea)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build/cartopy/tests/crs/test_lambert_azimuthal_equal_area.py", line 41, in test_default
    decimal=4)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 521, in assert_almost_equal
    return assert_array_almost_equal(actual, desired, decimal, err_msg)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 916, in assert_array_almost_equal
    precision=decimal)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 737, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 4 decimals

(mismatch 100.0%)
 x: array([-12727770.6158,  12727770.6158])
 y: array([-12727770.5987,  12727770.5987])

----------------------------------------------------------------------
```

It builds and tests fine on 64-bit. A detailed build log is available [here](http://debomatic-i386.debian.net/distribution#unstable/python-cartopy/0.14.2-1/buildlog).

**Comment by dopplershift** [comment]
Sounds like a double vs. float issue...

**Comment by QuLogic** [comment]
Weird, I'm sure I've been able to [build on 32-bit](https://copr.fedorainfracloud.org/coprs/qulogic/SciTools/build/179815/).

**Comment by ghisvail** [comment]
@QuLogic are there any tests being run in the copr builds? I can't see any explicit call to nose in the [spec](http://copr-dist-git.fedorainfracloud.org/cgit/qulogic/SciTools/python-cartopy.git/tree/python-cartopy.spec?id=0cb65e8b0297d241e691d4e5f4f054bde6569829) file and can't access the [log](http://copr-dist-git.fedorainfracloud.org/per-task-logs/179815-f24.log) either to verify.

Just to be clear, the build of the module runs fine but testing fails, resulting overall in a failure to build the package.

**Comment by QuLogic** [comment]
Oh, I though it was running tests, but apparently not. Maybe I'll try that out over the weekend.

**Comment by ghisvail** [comment]
Issue confirmed on the current Debian package [here](https://buildd.debian.org/status/package.php?p=python-cartopy).

For both `i386` and `hurd-i386`, the following error is thrown:
```
======================================================================
FAIL: test_default (cartopy.tests.crs.test_lambert_azimuthal_equal_area.TestLambertAzimuthalEqualArea)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«BUILDDIR»/python-cartopy-0.14.2+dfsg1/.pybuild/pythonX.Y_2.7/build/cartopy/tests/crs/test_lambert_azimuthal_equal_area.py", line 41, in test_default
    decimal=4)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 523, in assert_almost_equal
    return assert_array_almost_equal(actual, desired, decimal, err_msg)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 918, in assert_array_almost_equal
    precision=decimal)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 739, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 4 decimals

(mismatch 100.0%)
 x: array([-12727770.6158,  12727770.6158])
 y: array([-12727770.5987,  12727770.5987])

```

**Comment by ghisvail** [comment]
@QuLogic did you get an opportunity to try it out? Seems all 32-bit architectures are affected by this one test failure.

**Comment by jrtc27** [comment]
It's not a 32-bit issue; ARM, MIPS, PowerPC, Motorola 68000 and HP PA-RISC can all build it fine. It's an i386-only issue, presumably because Debian compiles without SSE2 support, forcing GCC to use the x87 FPU, which uses 80-bit floats internally, with the extra precision causing minor changes to the final value.

**Comment by ghisvail** [comment]
Aaron Ucko suggested to compile with `-ffloat-store` in the original bug report, which is not working.

**Comment by marqh** [comment]
given the range of considerations on this and that it appears to be only one test causing a problem, may I suggest that we look at a platform specific skip pattern?

Is there a neat way to identify that we are running on 32bit i386 and skip this test if that is the case.  the numerical difference is noticeable, but not vast

it could give us just enough to get this issue moved on for hte upcoming release

@ghisvail how would this feel to you?

**Comment by ghisvail** [comment]
> it could give us just enough to get this issue moved on for hte upcoming release

We are looking into it. It may be that it is the Debian version of proj4 which is the culprit, based on some further testing done by Aaron.

**Comment by marqh** [comment]
> We are looking into it. It may be that it is the Debian version of proj4 which is the culprit, based on some further testing done by Aaron.

on that basis, i think we will look to cut this release.

if it turns out that there is a small bug to fix, we can run out a 0.15.1 with it in in pretty short order

**Comment by ghisvail** [comment]
It was confirmed that our build of `proj4` is the problem. I have pushed a Debian specific fix to detect whether we are running on `i386` and skip the offending test if it that is the case. Closing.
